homeassistant/components/enasolar/config_flow.py

```diff
@@ -0,0 +1,202 @@
+"""EnaSolar solar inverter configuration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+import pyenasolar
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.core import callback
+import homeassistant.helpers.config_validation as cv
+from homeassistant.util import dt as dt_util
+
+from .const import (
+    CAPABILITY,
+    CONF_CAPABILITY,
+    CONF_DC_STRINGS,
+    CONF_HOST,
+    CONF_MAX_OUTPUT,
+    CONF_NAME,
+    CONF_SUN_DOWN,
+    CONF_SUN_UP,
+    DC_STRINGS,
+    DEFAULT_HOST,
+    DEFAULT_NAME,
+    DEFAULT_SUN_DOWN,
+    DEFAULT_SUN_UP,
+    DOMAIN,
+    MAX_OUTPUT,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class EnaSolarConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """User Configuration of EnaSolar Integration."""
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Initialize the config flow."""
+        self._enasolar = pyenasolar.EnaSolar()  # Only exposes methods, no significant
+        # code is executed during instantiation
+        self._data: dict[str, Any] = {}
+        self._errors: dict[str, str] = {}
+
+    def _conf_for_inverter_exists(self, serial) -> bool:
+        """Return True if inverter exists in configuration."""
+        for entry in self._async_current_entries(include_ignore=False):
+            if serial == entry.unique_id:
+                return True
+        return False
```

Comment:
```suggestion
        return any(
            entry
            for entry in self._async_current_entries(include_ignore=False)
            if serial == entry.unique_id
        )
```

Doesn't save any LOC, but could be written as an `any`

Reply:
As a python newbie, is your suggestion more pythonese? Is it the way one would normally implement this sort of logic ?

Reply:
I think this one is a matter of personal style choice.  I have a strong preference to all and any because it makes it easier to review since the intent is clear and it's harder to make a logic reversal error